norfair/metrics.py

```diff
@@ -0,0 +1,327 @@
+        self.matrixes_predictions = []
+        self.paths = []
+
+    def create_accumulator(self, input_path, information_file=None):
+
+        file_name = os.path.split(input_path)[1]
+
+        self.frame_number = 1
+        # save the path of this video in a list
+        self.paths = np.hstack((self.paths, input_path))
+        # initialize a matrix where we will save our predictions for this video (in the MOTChallenge format)
+        self.matrix_predictions = []
+
+        # initialize progress bar
+        if information_file is None:
+            seqinfo_path = os.path.join(input_path, "seqinfo.ini")
+            information_file = InformationFile(file_path = seqinfo_path)
+        length = information_file.search(variable_name = "seqLength")
+        self.progress_bar_iter = track(
+            range(length - 1), description=file_name, transient=False
+        )
+
+    def update(self, predictions=None):
+        # get the tracked boxes from this frame in an array
+        for obj in predictions:
+            new_row = [
+                self.frame_number,
+                obj.id,
+                obj.estimate[0, 0],
+                obj.estimate[0, 1],
+                obj.estimate[1, 0] - obj.estimate[0, 0],
+                obj.estimate[1, 1] - obj.estimate[0, 1],
+                -1,
+                -1,
+                -1,
+                -1,
+            ]
+            if np.shape(self.matrix_predictions)[0] == 0:
+                self.matrix_predictions = new_row
+            else:
+                self.matrix_predictions = np.vstack((self.matrix_predictions, new_row))
+        self.frame_number += 1
+        # advance in progress bar
+        try:
+            next(self.progress_bar_iter)
+        except StopIteration:
+            self.matrixes_predictions.append(self.matrix_predictions)
+            return
+
+    def compute_metrics(
+        self,
+        metrics=None,
+        generate_overall=True,
+    ):
+        if metrics is None:
+            metrics = list(mm.metrics.motchallenge_metrics)
+
+        self.summary = eval_motChallenge(
+            matrixes_predictions=self.matrixes_predictions,
+            paths=self.paths,
+            metrics=metrics,
+            generate_overall=generate_overall,
+        )
+
+        return self.summary
+
+    def save_metrics(self, save_path=".", file_name="metrics.txt"):
+
+        if not os.path.exists(save_path):
+            os.makedirs(save_folder)
+        metrics_path = os.path.join(save_path, file_name)
+        metrics_file = open(metrics_path, "w+")
+        metrics_file.write(self.summary)
+        metrics_file.close()
+
+    def print_metrics(self):
+        print(self.summary)
+
+
+def load_motchallenge(matrix_data, min_confidence=-1):
+    """Load MOT challenge data.
+
+    This is modification of the function load_motchallenge from the MOT library, defined in io.py
+    The panda dataframe is generated from a numpy array (matrix_data) instead of a text file.
```

Comment:
Small gramatical mistake in 'is **a** modification of...'. Please use the full name of the library not 'the MOT library', I think the name was 'py-motmetrics'?. Panda**s** has an s at the end. I would also start the second sentence with `In this version of the file, the pandas dataframe blah blah blah`.